zerver/tests/test_signup.py

```diff
@@ -1740,6 +1741,15 @@ def test_create_realm_no_creation_key(self) -> None:
             self.assertEqual(result.status_code, 200)
             self.assert_in_response('New organization creation disabled', result)
 
+    @override_settings(TERMS_OF_SERVICE=False)
+    def test_create_dev_realm(self) -> None:
+        count = UserProfile.objects.count()
+        string_id = "realm-%d" % (count,)
+        result = self.client_post('/register_realm/')
+        realm = Realm.objects.get(string_id=string_id)
+        self.assertEqual(result.status_code, 302)
+        self.assertEqual(realm.string_id, string_id)
```

Comment:
I'd verify the URL in `result` (I think `result.location`?) as well as that the user is logged in (`test_signup` has a bunch of examples of how to do this).

Reply:
Similarly with the other redirects.